coreneuron/io/reports/report_handler.cpp

```diff
@@ -155,6 +161,73 @@ VarsToReport ReportHandler::get_compartment_vars_to_report(const NrnThread& nt,
     return vars_to_report;
 }
 
+std::string getSectionTypeStr(SectionType type) {
+    switch (type) {
+        case Axon:
+            return "axon";
+        case Dendrite:
+            return "dend";
+        default:
+            return "apic";
+    }
+}
+
+VarsToReport ReportHandler::get_section_vars_to_report(const NrnThread& nt,
+                                                       const std::set<int>& target,
+                                                       double* report_variable,
+                                                       SectionType section_type,
+                                                       bool all_compartments) const {
+    VarsToReport vars_to_report;
+    const auto section_type_str = getSectionTypeStr(section_type);
```

Comment:
bind temporary to const reference
```suggestion
    const auto& section_type_str = getSectionTypeStr(section_type);
```